server/src/main/java/org/opensearch/index/recovery/RemoteStoreRestoreService.java

```diff
@@ -233,6 +251,29 @@ private RemoteRestoreResult executeRestore(
         return RemoteRestoreResult.build(restoreUUID, restoreInfo, allocationService.reroute(updatedState, "restored from remote store"));
     }
 
+    private ClusterState restoreClusterMetadata(ClusterState restoredClusterState, Metadata clusterMetadata) {
+        Metadata.Builder mdBuilder = new Metadata.Builder(clusterMetadata);
+        mdBuilder.clusterUUID(restoredClusterState.getMetadata().clusterUUID());
+        mdBuilder.clusterUUIDCommitted(restoredClusterState.getMetadata().clusterUUIDCommitted());
+        mdBuilder.coordinationMetadata(restoredClusterState.getMetadata().coordinationMetadata());
+        if (clusterMetadata.persistentSettings() != null) {
+            Settings settings = clusterMetadata.persistentSettings();
+            clusterService.getClusterSettings().validateUpdate(settings);
+            mdBuilder.persistentSettings(settings);
+        }
+        Optional<RepositoriesMetadata> repositoriesMetadata = Optional.ofNullable(clusterMetadata.custom(RepositoriesMetadata.TYPE));
+        repositoriesMetadata = repositoriesMetadata.map(
+            repositoriesMetadata1 -> new RepositoriesMetadata(
+                repositoriesMetadata1.repositories()
+                    .stream()
+                    .filter(repository -> SYSTEM_REPOSITORY_SETTING.get(repository.settings()) == false)
+                    .collect(Collectors.toList())
+            )
+        );
+        repositoriesMetadata.ifPresent(metadata -> mdBuilder.putCustom(RepositoriesMetadata.TYPE, metadata));
+        return ClusterState.builder(restoredClusterState).metadata(mdBuilder).build();
```

Comment:
We are now getting Metadata object from RemoteClusterStateService comprised of both Index/Global Metadata. So here `.metadata(mdBuilder)` will override the IndexMetadata which we have restored in correct format in `restoredClusterState` and it will break the restoration of IndexMetadata.

We might need remove indices from Metadata object which we are receiving in this method, before using here.

Also can we combine the restoration of IndexMetadata/GlobalMetadata in single method? I think that will be easy and we don't need to worry about such cases.